controllers/utils/utils.go

```diff
@@ -295,3 +296,23 @@ func GetField(values map[string]string, key, defaultValue string) string {
 	}
 	return defaultValue
 }
+
+func GetPod(ctx context.Context, client client.Client, ns string) (*corev1.Pod, error) {
```

Comment:
I'd be careful regarding (I presume) copy/pasting code from third-party libraries and license.

Just do the client.Get(... pod) command in GetDeployment. The rest here is not relevant for us.